lib/chef/resource/chef_client_config.rb

```diff
@@ -250,20 +250,21 @@ def string_to_symbol(prop_val)
         introduced: "17.8"
 
       action :create, description: "Create a client.rb config file for configuring #{ChefUtils::Dist::Infra::PRODUCT}." do
-        unless ::Dir.exist?(new_resource.config_directory)
-          directory new_resource.config_directory do
+        [
+          new_resource.config_directory,
+          (::File.dirname(new_resource.log_location) unless new_resource.log_location.nil?),
+          new_resource.file_backup_path,
+          new_resource.file_cache_path,
+          ::File.join(new_resource.config_directory, "client.d"),
+          (::File.dirname(new_resource.pid_file) unless new_resource.pid_file.nil?),
+        ].each do |dir_path|
+          next if dir_path.nil?
+          next if ::Dir.exist?(dir_path)
```

Comment:
can you change this to just use `].compact.each do |dir_path|` and then drop both of the checks since `compact` will remove the nils and we should rely on the directory resource's idempotency check and not manually do that check (the original code should not have been doing that check either).

Reply:
(the action on the directory resource may need to be updated to `:create_if_missing`?  although i think that it should actively manage the file perms anyway since we'd want to revert any changes on the node that made them 0777 for example)

Reply:
changes made. left out :create_if_missing as managing user, group, mode if manually changed does seem like the right thing to do.